### PR TITLE
Fixes #1046:Query does not change from news tab to any other tab

### LINKED ISSUE #1046: Query is changed when switching from news tab to any other results tab.
**I'm submitting a ...**
- [x] bug report
- [ ] feature request

**Current behavior:**

Currently, query is changed when switching from news tab to any other results tab.
**Expected behavior:**

It should not change.

### PR DESCRIPTION
**Pull request by praveenojha33**
Fixes #1046 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `master` branch.
- [x] I have added necessary documentation (if appropriate)
- [x] Added Surge preview link

#### Changes proposed in this pull request:
Query does not change from news tab to any other tab
Surge Link:https://pr-1047-fossasia-susper.surge.sh

### CODE REVIEW

**Comment by simsausaurabh** [comment]
@praveenojha33 The surge link stops working when I click on news and then back to any other tab.

File: src/app/results/results.component.ts [hunk 1]
```diff
@@ -212,6 +212,9 @@ export class ResultsComponent implements OnInit {
       urldata.rows = Number(query['rows']) || 10;
       this.presentPage = Math.abs(query['start'] / urldata.rows) + 1;
       let querydata = Object.assign({}, urldata);
+      if (this.resultDisplay !== 'news') {
```
Comment (simsausaurabh) [inline src/app/results/results.component.ts]: Please explain why are you doing this. It seems a mere workaround.

**Comment by praveenojha33** [comment]
@simsausaurabh The changes are very less you can use your local installation to check it.

File: src/app/results/results.component.ts (hunk 1 above)
Comment (praveenojha33) [reply src/app/results/results.component.ts]: Actually it is not a workaround, it is a fix. If we are not in news tab we will replace ` site:www.dailymail.co.uk` to ''
` site:` is used after a query to filter all results from that site.

Comment (simsausaurabh) [reply src/app/results/results.component.ts]: Can you create an action and effect to keep news search independent from normal search queries? In this way you would not need to replace the site query explicitly and both the searching will be independent and each time user switches from news tab and other tabs, it won't affect other.

Comment (praveenojha33) [reply src/app/results/results.component.ts]: Yes I am working on it but till then it will be a fix.

Comment (praveenojha33) [reply src/app/results/results.component.ts]: Please review the changes and tell if it has any limitation or any problem it causes.

**Comment by simsausaurabh** [comment]
@praveenojha33 after running it locally, I have found out few issues please see:

1. When I search for any query and switch to images or videos tab, the resultant URL changes to the value which is different from that expected.

screenshot from 2018-06-26 22-54-16

2. After searching any query, when I click on news tab then the query changes in input box.

screenshot from 2018-06-26 22-54-44

3. After we get results, If I click on news tab and then go back to main page of susper from clicking top left logo, then query remains same as it is appeared in above issue (query <space> site:<site-of-news>).

screenshot from 2018-06-26 22-55-55

**Comment by praveenojha33** [comment]
@hongquan It should change, to show results what is problem with that?
Is changing URL causing any issue?
See google.com there also URL changes when you move to any other tab

**Comment by simsausaurabh** [comment]
@praveenojha33 please see comments again. It is not an issue in changing URL but it is not what is expected according to your implementation (See the URL in first image before ```mode=text```, the value which is in ````()`` brackets should not be it. Right?). And for second point, should it be ```google site:www.dailymail.co.uk```? And the third issue is dangerous, it is actually changing the displayed text in input search field which should not be there.

**Comment by praveenojha33** [comment]
@simsausaurabh everything is right about URL it is done to sort out results please have a look here https://lucene.apache.org/solr/guide/6_6/common-query-parameters.html . Also the query changes whenever we go to news tab to filter results from that site and it is restored again whenever we switch to other Tab. I am working on a proper implementation of News Tab after which this issue will not be there. Thank You

**Comment by simsausaurabh** [comment]
Okay, It clears doubt about the first one since it seems that URL is being updated slowly (on complete load). But you need to work on the issue-2 (query changing in news tab input) and issue-3 (query of news tab on home page) in further patches.

Rest LGTM :+1:

**Comment by praveenojha33** [comment]
Changes in this PR are no longer needed.

**Pull request closed without merging**